frontend/src/pages/pipelines/global/pipelines/PipelinesPageHeaderActions.tsx

```diff
@@ -1,39 +1,76 @@
 import * as React from 'react';
-import { Dropdown, DropdownItem, DropdownSeparator, DropdownToggle } from '@patternfly/react-core';
+import {
+  Dropdown,
+  DropdownItem,
+  DropdownSeparator,
+  MenuToggle,
+  Tooltip,
+} from '@patternfly/react-core';
 import { DeleteServerModal, ViewServerModal } from '~/concepts/pipelines/context';
 
-const PipelinesPageHeaderActions: React.FC = () => {
+type PipelinesPageHeaderActionsProps = {
+  toggleElement: React.ReactNode;
+  variant: 'plain' | 'secondary';
+  isDisabled: boolean;
+};
+
+const PipelinesPageHeaderActions: React.FC<PipelinesPageHeaderActionsProps> = ({
+  toggleElement,
+  variant,
+  isDisabled,
+}) => {
   const [open, setOpen] = React.useState(false);
   const [deleteOpen, setDeleteOpen] = React.useState(false);
   const [viewOpen, setViewOpen] = React.useState(false);
+  const onToggleClick = () => {
+    setOpen(!open);
+  };
```

Comment:
Help me understand why you took this out of the React render flow? What benefit does this have being a stand alone function? It's only used once -- your "DropdownItem" has an inline `onClick` 🤔 

Is there a flow to which you write your functions that just leads it to be separate and since there is no harm you leave it?

FWIW... Not that it usually matters for `onClick` handlers or even in this use-case -- but separating functions from the component prop callbacks serves to remove the types 🤔 If you used this function twice or managed the prop state it would make more sense. But I am worried about scenarios where you need a return type or props typed, what would you do in those cases?

Reply:
I've inlined this now.

Reply:
@manaswinidas Thanks! Could you help me understand the why though?

Reply:
I did this for modularity. But a one-liner doesn't quite affect the modularity much. It negatively affects the readability.

Reply:
> It negatively affects the readability.

I am not sure pulling out the inline actions of buttons and similar components is good for readability -- it breaks up the flow. This makes a split reading problem. Reading JSX top to bottom to understand the structure but once I hit a "what does the button click do" you're calling another function causing me to have to "leave the JSX" to understand the implementation -- this adds an unnecessary split in reading.

I've worked a lot on JSX and I have rarely found a problem with inline button functionality, especially for a toggle (1-liner) action, causing an issue. Often times when it happens it's because you're overloading the component's actions with a ton of operations. If you need to do lots of business logic on an action, you should consider what that means for the component -- if the log is too much for inline, then you probably are violating a lot of the effort of what the component is intended to do. Consider splitting up with Hooks or creating new components to manage this logic. 

This is a common problem I find with the team is the separation of the logic. I'll need to write docs to explain what this means and how breaking stuff up into declarative chunks helps with readability.

Reply:
I meant Pulling out the inline actions negatively affects the readability.